## Mapping

Once the reads have been quality checked and trimmed, the next step is to map the 
reads to the reference genome (in our case the human genome "hg19"). This can be 
done with the Bioconductor package `Rsubread` \cite{Liao2013}.

Before mapping the reads to the reference genome you will need to build a
Rsubread index for that genome. Below are the commands for building an index for 
the human reference genome using the `buildindex` command.

```{r setup-filepaths,eval=F}
library(Rsubread)
RNASeq_REF_DATA_DIR <- "../data/Data_Analysis_with_R/RNAseq/ref_data/"

REF_GENOME <- file.path(RNASeq_REF_DATA_DIR, "hg19.fa")

RSUBREAD_INDEX_PATH <- file.path(RNASeq_REF_DATA_DIR, "RsubreadIndex")

RSUBREAD_INDEX_BASE <- "hg19"
```

```
buildindex(basename=file.path(RSUBREAD_INDEX_BASE, RSUBREAD_INDEX_PATH),
           reference=REF_GENOME)
```

Once the Rsubread index has been created you can map your reads to the genome by 
running the `align` command. The code below could be used to map the reads for 
a specific library against the "hg19" genome.

```{block, type="rmdtip"}
Fastq files need to be unzipped first.
```

```{r}
sample <- "ERR420386"

inputFileFWD <- file.path(RNASeqDATADIR, paste0(sample,"_1.fastq"))
inputFileRVS <- file.path(RNASeqDATADIR, paste0(sample,"_2.fastq"))

MAPPING_DIR <- "../data/Data_Analysis_with_R/RNAseq/mapping"

output.samFile <- file.path(MAPPING_DIR, paste0(sample,".sam"))
output.bamFile <- file.path(MAPPING_DIR, paste0(sample,".bam"))
```

```{block, type="rmdwarning"}
**NOTE**: For the purpose of this course the mapping has already been done. 
This step can take up to a couple of hours per library. 
**Please do not run the command below.**
```

```
align(index = file.path(RSUBREAD_INDEX_PATH, RSUBREAD_INDEX_BASE),
     readfile1 = inputFileFWD, 
     readfile2 = inputFileRVS,
     output_file = output.samFile)
```

**CONVERT TO BAM FILE**

```{block, type="rmdtip"}
The `nthreads` parameter can be used in the align command to speed up teh
process and run the alignment using several CPUs in parallel.
```

The function `propmapped` returns the proportion of mapped reads in the output 
SAM file: total number of input reads, number of mapped reads and proportion of mapped reads.

```{r}
propmapped(output.samFile)
```

Rsubread provides a read summarization function `featureCounts`, which takes as
input the SAM or BAM files and assigns them to genomic features. This gives the 
number of reads mapped per gene, which can then be transformed into RPKM values 
(Read Per Killobase per Million), normalised and tested for differential expression. 

In-built annotations for mm9, mm10 and hg19 are provided for users' convenience, 
or you can also use the index previously built.

Getting read counts using the inbuilt hg19 annotation
```{r}
counts <- featureCounts(output.bamFile, 
                        annot.inbuilt="hg19",
                        isGTFAnnotationFile=FALSE, 
                        isPairedEnd=TRUE)

#Get the number of reads mapped per feature
summary(counts)
```


```{block, type="rmdexercise"}
Run the above command to perform counting reads for the bam file and then take
a look at the summary output.
```


```{block, type="rmdtip"}
If you want to get read counts using the index previously built, use the
following command:
```counts <- featureCounts(output.samFile, 
                        annot.ext=file.path(RSUBREAD_INDEX_PATH, RSUBREAD_INDEX_BASE),
                        isGTFAnnotationFile=TRUE,
                        isPairedEnd=TRUE)
```
```


**NOTE** For the purpose of this course the read summarisation step has already 
been performed for all libraries. You will need to load the corresponding RData 
file to get these read counts. You can then print out these counts in a text 
file for future reference.

To load the Rdata object file:
```{r}
load(file.path(MAPPING_DIR, "RNAseqCounts.RData"))

summary(counts)
```

```{r}
counts$targets
```

Export out the counts table for every sample into a tab-separated file:
```{r}
write.table(counts$counts, file="raw_read_counts.txt", sep="\t", quote=F, append=F)
```
