## Differential expression analysis

### Fitting the model

To fit the model, use the `lmFit()` function, which takes in the normalised data object and the design matrix:
```{r}
fit <- lmFit(y,design)
```

### Comparisons

To have a look at the pairwise comparison(s) of interest, we may need to create a constrast matrix. This is usually always the case when there are multiple factors being compared. To do this, we use the `makeConstrasts()` function:

```{r constrast-matrix}
cont.matrix <- makeContrasts(liver-brain, levels=design)
cont.matrix
```

Refit the model using the comparisons defined:
```{r contrast-fit}
fit2 <- contrasts.fit(fit, cont.matrix)
fit2 <- eBayes(fit2)
options(digits = 3)

dim(fit2)
```

```{r, echo=F}
save(fit,fit2,design,cont.matrix, file='results/RNAseq/limmaDEG.Rdata')
```


### Extract top DE features
The `topTable` function summarises the output from limma in a table format. 
Significant DE genes for a particular comparison can be identified by selecting 
genes with a p-value smaller than a chosen cut-off value and/or a fold change 
greater than a chosen value in this table. By default the table will be sorted
by increasing adjusted p-value, showing the most significant DE genes at the top.

Set the threshold values:
```{r threshold-setting}
THRES_PVAL <- 0.01
THRES_FC <- 3

colnames(fit$coefficients)
```

Get the output table for the 10 most significant DE genes for this comparison: 
$liver vs brain$.
```{r toptable}
comparison='liver - brain'
topTable(fit2, coef=comparison)
```

Get the full table (`n` is the number of genes in the fit):
```{r full-toptable}
limma.result <- topTable(fit2, coef=comparison, n=nrow(fit2))

## Get significant DEGs only (adjusted p-value < THRES_PVAL)
limma.sigP.DEG <- topTable(fit2, coef=comparison, n=nrow(fit2), p.val=THRES_PVAL)
dim(limma.sigP.DEG)
head(limma.sigP.DEG,10)
```

Get significant DEG with low adjusted p-values and high fold change
```{r sigDEG-thresholds}
limma.sigFC.DEG <- subset(limma.sigP.DEG, logFC > THRES_FC)
dim(limma.sigFC.DEG)
head(limma.sigFC.DEG,10)
```
```{r, echo=F}
save(limma.sigFC.DEG, limma.sigP.DEG, file=file.path(RESULTS_DIR,'limma.sig.Rdata'))
```


Write the limma output table for significant genes to a tab-delimited file:
```{r sigDEG-output}
filename <- paste0("DEG_",comparison,
                   "_pval_",THRES_PVAL,
                   "_logFC_", THRES_FC, ".txt");
write.table(limma.sigFC.DEG, file=file.path(RESULTS_DIR,filename),
            row.names=T, quote=F, sep="\t")
filename
```

```{block, type='rmdexercise'}
Get the number of DE genes between technical group 1 and technical group 2 (all Brain samples) with adj pvalue<0.01

* Create a new design matrix for limma with the technical replicate groups
* Re-normalise the read counts with 'voom' function with new design matrix
* Fit a linear model on these normalised data
* Make the contrast matrix corresponding to the new set of parameters
* Fit the contrast matrix to the linear model
* Compute moderated t-statistics of differential expression 
* Get the output table for the 10 most significant DE genes for this comparison
* Get the number of genes significantly DE (adjusted p-value < 0.01)
```

## Verification using visualisation

Plot the top 6 DEGs to verify that they are indeed different between the groups $brain vs liver$.

The `tidyr` library helps us reshape the data from the wide form into a long form, which is much more flexible to work with when using `ggplot` for plotting graphs.

```{r verification}
library(tidyr)
topDEG <- rownames(limma.sigFC.DEG)[1:6]
topDEG.norm <- as.data.frame(norm.expr[which(rownames(norm.expr) %in% topDEG),])
topDEG.norm$geneID <- rownames(topDEG.norm)
topDEG.norm.long <- gather(topDEG.norm, key=sample, value=value, -geneID)
topDEG.norm.long$group <- expr.design[topDEG.norm.long$sample,'tissue']

ggplot(topDEG.norm.long) + geom_point(aes(group,value,col=group),size=2,pch=1) + 
  theme_bw() + facet_wrap(~geneID)
```

### Hierachical clustering

In order to investigate the relationship between samples, hierarchical clustering 
can be performed using the `heatmap` function. In this example, `heatmap` calculates 
a matrix of euclidean distances from the normalised expression for the 100 most signficant DE genes.

```{r histogram-clustering, fig.height=4}
topDEG <- rownames(limma.sigFC.DEG)[1:100]
highNormGenes <- norm.expr[topDEG,]
dim(highNormGenes)
par(cex.main=1) 
heatmap(highNormGenes, col=topo.colors(50), cexCol=1,
        main='Top 100 DEG')
```

You will notice that the samples clustering does not follow the original order 
in the data matrix (alphabetical order "ERR420386" to "ERR420393"). They have 
been re-ordered according to the similarity of the 100 genes expression profiles.
To understand what biological effect lies under this clustering, one can use the 
samples annotation for labeling (samples group, age, sex etc).

```{r hist-cluster-group, fig.height=4}
par(cex.main=1) 
heatmap(highNormGenes, col=topo.colors(50),cexCol=1,
        main='Top 100 DEG', labCol = group)
```

```{block, type="rmdexercise"}
Produce a heatmap for the bottom 100 significant genes.

* How many "groups" do you see?
* Can you explain them with the experimental design?
```

```{r bottom-DGE, echo=F, fig.height=4}
bottomDEG <- tail(rownames(limma.sigFC.DEG),100)
par(cex.main=1) 
heatmap(norm.expr[bottomDEG,],col=topo.colors(50),cexCol=1,
        main="Bottom 100 DEG")
```

```{block, type='rmdexercise'}
**Challenge**

* Just to be extra sure and for our own confidence, randomly select another 100 genes that are not in the DEG list and repeat the hierachical clustering using the `heatmap` plot.

* Out of these 100, pick 6 to plot the gene wise differences as seen previously.
```

```{block, type='rmdsolution'}
**Solution**

\begin{verbatim}
nonDEG <- setdiff(rownames(filtered.raw.counts),rownames(limma.sigFC.DEG))
length(nonDEG)
randGene <- sample(nonDEG, 100, replace=F)
heatmap(norm.expr[randGene,],col=topo.colors(50),main="Non DEG",cexCol=1)
\end{verbatim}
```



```{r nonDGE, include=F, fig.height=4}
nonDEG <- setdiff(rownames(filtered.raw.counts),rownames(limma.sigFC.DEG))
length(nonDEG)
randGene <- sample(nonDEG, 100, replace=F)
heatmap(norm.expr[randGene,],col=topo.colors(50),main="Non DEG",cexCol=1)
```

```{r, include=F}
gene.wise <- as.data.frame(norm.expr[randGene[1:6],])
gene.wise$geneID <- rownames(gene.wise)
gene.wise <- gather(gene.wise, key=sample, value=value, -geneID)
gene.wise$group <- expr.design[gene.wise$sample,'tissue']
gene.wise

ggplot(gene.wise) + geom_point(aes(group,value,col=group),size=2,pch=1) + 
  theme_bw() + facet_wrap(~geneID)
```

