<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Differentially expressed gene analysis of RNA-seq data using R</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Differentially expressed gene analysis of RNA-seq data using R">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="normalisation.html">
<link rel="next" href="differential-expression-analysis.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> NGS data exploration</a><ul>
<li class="chapter" data-level="1.1" data-path="required-packages-for-these-exercises.html"><a href="required-packages-for-these-exercises.html"><i class="fa fa-check"></i><b>1.1</b> Required packages for these exercises</a></li>
<li class="chapter" data-level="1.2" data-path="reading-from-a-fasta-file.html"><a href="reading-from-a-fasta-file.html"><i class="fa fa-check"></i><b>1.2</b> Reading from a fasta file</a></li>
<li class="chapter" data-level="1.3" data-path="parsing-fastq-files.html"><a href="parsing-fastq-files.html"><i class="fa fa-check"></i><b>1.3</b> Parsing FastQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rnaseq-data-analysis.html"><a href="rnaseq-data-analysis.html"><i class="fa fa-check"></i><b>2</b> RNAseq data analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="mapping.html"><a href="mapping.html"><i class="fa fa-check"></i><b>2.2</b> Mapping</a><ul>
<li class="chapter" data-level="2.2.1" data-path="mapping.html"><a href="mapping.html#examining-the-mapped-reads"><i class="fa fa-check"></i><b>2.2.1</b> Examining the mapped reads</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>2.3</b> Quantification</a></li>
<li class="chapter" data-level="2.4" data-path="quality-checking.html"><a href="quality-checking.html"><i class="fa fa-check"></i><b>2.4</b> Quality checking</a></li>
<li class="chapter" data-level="2.5" data-path="checking-for-batch-or-confounding-effects.html"><a href="checking-for-batch-or-confounding-effects.html"><i class="fa fa-check"></i><b>2.5</b> Checking for batch or confounding effects</a></li>
<li class="chapter" data-level="2.6" data-path="prefiltering.html"><a href="prefiltering.html"><i class="fa fa-check"></i><b>2.6</b> Prefiltering</a></li>
<li class="chapter" data-level="2.7" data-path="normalisation.html"><a href="normalisation.html"><i class="fa fa-check"></i><b>2.7</b> Normalisation</a></li>
<li class="chapter" data-level="2.8" data-path="batch-and-sanity-checks.html"><a href="batch-and-sanity-checks.html"><i class="fa fa-check"></i><b>2.8</b> Batch and sanity checks</a><ul>
<li class="chapter" data-level="2.8.1" data-path="batch-and-sanity-checks.html"><a href="batch-and-sanity-checks.html#hierachical-clustering"><i class="fa fa-check"></i><b>2.8.1</b> Hierachical clustering</a></li>
<li class="chapter" data-level="2.8.2" data-path="batch-and-sanity-checks.html"><a href="batch-and-sanity-checks.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>2.8.2</b> Principal Component Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>2.9</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.9.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#model-matrices"><i class="fa fa-check"></i><b>2.9.1</b> Model matrices</a></li>
<li class="chapter" data-level="2.9.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#ma-plots"><i class="fa fa-check"></i><b>2.9.2</b> MA-plots</a></li>
<li class="chapter" data-level="2.9.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#fitting-the-model"><i class="fa fa-check"></i><b>2.9.3</b> Fitting the model</a></li>
<li class="chapter" data-level="2.9.4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#comparisons"><i class="fa fa-check"></i><b>2.9.4</b> Comparisons</a></li>
<li class="chapter" data-level="2.9.5" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#extract-top-de-features"><i class="fa fa-check"></i><b>2.9.5</b> Extract top DE features</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="verification.html"><a href="verification.html"><i class="fa fa-check"></i><b>2.10</b> Verification</a></li>
<li class="chapter" data-level="2.11" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>2.11</b> Gene Annotation</a></li>
<li class="chapter" data-level="2.12" data-path="gene-set-enrichment.html"><a href="gene-set-enrichment.html"><i class="fa fa-check"></i><b>2.12</b> Gene set enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sessioninfo.html"><a href="sessioninfo.html"><i class="fa fa-check"></i><b>3</b> SessionInfo</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differentially expressed gene analysis of RNA-seq data using R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="batch-and-sanity-checks" class="section level2">
<h2><span class="header-section-number">2.8</span> Batch and sanity checks</h2>
<div id="hierachical-clustering" class="section level3">
<h3><span class="header-section-number">2.8.1</span> Hierachical clustering</h3>
<p>In order to investigate the relationship between samples, hierarchical clustering can be performed using the <code>heatmap</code> function. In this example, <code>heatmap</code> calculates a matrix of euclidean distances from the mapped read counts for the 100 most highly expressed genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">select &lt;-<span class="st"> </span><span class="kw">order</span>(<span class="kw">rowMeans</span>(counts$counts), <span class="dt">decreasing=</span><span class="ot">TRUE</span>)[<span class="dv">1</span>:<span class="dv">100</span>]
highexprgenes &lt;-<span class="st"> </span>counts$counts[select,]

<span class="kw">heatmap</span>(highexprgenes, <span class="dt">col=</span><span class="kw">topo.colors</span>(<span class="dv">50</span>), <span class="dt">margin=</span><span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">6</span>), <span class="dt">cexCol=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="RNAseq-analysis_files/figure-html/histogram-clustering-1.png" width="672" /></p>
<p>You will notice that the samples clustering does not follow the original order in the data matrix (alphabetical order “ERR420386” to “ERR420393”). They have been re-ordered according to the similarity of the 100 genes expression profiles. To understand what biological effect lies under this clustering, one can use the samples annotation for labeling (samples group, age, sex etc).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">heatmap</span>(highexprgenes, <span class="dt">col=</span><span class="kw">topo.colors</span>(<span class="dv">50</span>), <span class="dt">margin=</span><span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">6</span>), <span class="dt">cexCol=</span><span class="dv">1</span>, <span class="dt">labCol =</span> group)</code></pre></div>
<p><img src="RNAseq-analysis_files/figure-html/hist-cluster-group-1.png" width="672" /></p>
<div class="rmdexercise">
<p>
<strong>Produce a heatmap for the 30 most highly expressed genes and annotate the samples with their age</strong>
</p>
<ul>
<li>
<p>
List the samples age from the experimental design (check order!)
</p>
</li>
<li>
<p>
Annotate the samples in the normalised dataset with their age
</p>
</li>
<li>
<p>
Subset the read counts object for teh 30 most highly expressed genes
</p>
</li>
<li>
<p>
Plot a heatmap with this subset of data, scaling genes and ordering both genes and samples
</p>
</li>
<li>
<p>
Increase or decrease the genes and samples labels font size as required
</p>
</li>
</ul>
</div>
</div>
<div id="principal-component-analysis-pca" class="section level3">
<h3><span class="header-section-number">2.8.2</span> Principal Component Analysis (PCA)</h3>
<p>A Principal Component Analysis (PCA) can also be performed with these data using the <code>mixOmics</code> package . The proportion of explained variance histogram will show how much of the variability in the data is explained by each components.</p>
<p>Reads counts need to be transposed before being analysed with the <code>mixomics</code> functions, i.e. genes should be in columns and samples should be in rows. This is the code for transposing and checking the data before further steps:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(mixOmics)

rawcounts.df &lt;-<span class="st"> </span><span class="kw">t</span>(counts$counts)
<span class="kw">dim</span>(rawcounts.df)</code></pre></div>
<pre><code>## [1]     8 25702</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## filter, remove features where it has 0 counts across all samples
zero.cols &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colSums</span>(rawcounts.df) ==<span class="st"> </span><span class="dv">0</span>)
rawcounts.filtered &lt;-<span class="st"> </span>rawcounts.df[,-zero.cols]
<span class="kw">dim</span>(rawcounts.filtered)</code></pre></div>
<pre><code>## [1]     8 23515</code></pre>
<p>The proportion of explained variance helps you determine how many components can explain the variability in your dataset and thus how many dimensions you should be looking at.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tuning &lt;-<span class="st"> </span><span class="kw">tune.pca</span>(rawcounts.filtered, <span class="dt">center=</span><span class="ot">TRUE</span>, <span class="dt">scale=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Eigenvalues for the first 8 principal components, see object$sdev^2: 
##          PC1          PC2          PC3          PC4          PC5 
## 1.163953e+04 5.804689e+03 3.047613e+03 8.260191e+02 7.918748e+02 
##          PC6          PC7          PC8 
## 7.788229e+02 6.264530e+02 1.209178e-25 
## 
## Proportion of explained variance for the first 8 principal components, see object$explained_variance: 
##          PC1          PC2          PC3          PC4          PC5 
## 4.949831e-01 2.468505e-01 1.296029e-01 3.512733e-02 3.367531e-02 
##          PC6          PC7          PC8 
## 3.312026e-02 2.664057e-02 5.142155e-30 
## 
## Cumulative proportion explained variance for the first 8 principal components, see object$cum.var: 
##       PC1       PC2       PC3       PC4       PC5       PC6       PC7 
## 0.4949831 0.7418336 0.8714365 0.9065639 0.9402392 0.9733594 1.0000000 
##       PC8 
## 1.0000000 
## 
##  Other available components: 
##  -------------------- 
##  loading vectors: see object$rotation</code></pre>
<p><img src="RNAseq-analysis_files/figure-html/tuning-1.png" width="672" /></p>
<p>The variable <code>tune$prop.var</code> indicates the proportion of explained variance for the first 10 principal components:</p>
<p>Plotting this variable makes it easier to visualise and will allow future reference.</p>
<p>In most cases, the first 2 or 3 components explain more than half the variability in the dataset and can be used for plotting. The <code>pca</code> function will perform a principal components analysis on the given data matrix. The <code>plotIndiv</code> function will provide scatter plots for sample representation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result &lt;-<span class="st"> </span><span class="kw">pca</span>(rawcounts.filtered, <span class="dt">ncomp=</span><span class="dv">3</span>, <span class="dt">center=</span>T, <span class="dt">scale=</span>T)
<span class="kw">plotIndiv</span>(result, <span class="dt">comp=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</code></pre></div>
<p><img src="RNAseq-analysis_files/figure-html/run-pca-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotIndiv</span>(result, <span class="dt">comp=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="dt">group=</span>group)</code></pre></div>
<p><img src="RNAseq-analysis_files/figure-html/sample-plot-group-1.png" width="672" /></p>
<p>The PCA plot of the first two components show a clear separation of the Brain and Liver samples across the 1st dimension. Within each sample group we can also notice a split between the 4 samples of each group, which seem to cluster in pair. This observation can be explained by another factor of variability in the data, commonly batch effect or another biological biais such as age or sex.</p>
<p><strong>DON’T NEED TO REDO THE PCA IF THE ORDER OF SAMPLES IS THE SAME, JUST COLOUR</strong></p>
<p><strong>SHOULD DO PCA ON THE WHOLE DATASET</strong></p>
<div class="rmdexercise">
<p>
For the 30 most highly expressed genes, we want to identify the reason for the split between samples from the same tissues. To do this, break the problem down:
</p>
<ul>
<li>
Get the read counts for the 30 most highly expressed genes
</li>
<li>
Transpose this matrix of read counts
</li>
<li>
Check the number of dimensions explaining the variability in the dataset
</li>
<li>
Run the PCA with an appropriate number of components
</li>
<li>
Annotate the samples with their age
<ul>
<li>
re-run PCA
</li>
<li>
plot the main components
</li>
</ul>
</li>
<li>
Annotate the samples with other clinical data
<ul>
<li>
re-run the PCA
</li>
<li>
plot the main components until you can separate the samples within each tissue group
</li>
</ul>
</li>
</ul>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="normalisation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-expression-analysis.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-QC_stats.Rmd",
"text": "Edit"
},
"download": ["RNAseq-analysis.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
