<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Differentially expressed gene analysis of RNA-seq data using R</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Differentially expressed gene analysis of RNA-seq data using R">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="rnaseq-data-analysis.html">
<link rel="next" href="mapping.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> NGS data exploration</a><ul>
<li class="chapter" data-level="1.1" data-path="required-packages-for-these-exercises.html"><a href="required-packages-for-these-exercises.html"><i class="fa fa-check"></i><b>1.1</b> Required packages for these exercises</a></li>
<li class="chapter" data-level="1.2" data-path="reading-from-a-fasta-file.html"><a href="reading-from-a-fasta-file.html"><i class="fa fa-check"></i><b>1.2</b> Reading from a fasta file</a></li>
<li class="chapter" data-level="1.3" data-path="parsing-fastq-files.html"><a href="parsing-fastq-files.html"><i class="fa fa-check"></i><b>1.3</b> Parsing FastQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rnaseq-data-analysis.html"><a href="rnaseq-data-analysis.html"><i class="fa fa-check"></i><b>2</b> RNAseq data analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="mapping.html"><a href="mapping.html"><i class="fa fa-check"></i><b>2.2</b> Mapping</a><ul>
<li class="chapter" data-level="2.2.1" data-path="mapping.html"><a href="mapping.html#examining-the-mapped-reads"><i class="fa fa-check"></i><b>2.2.1</b> Examining the mapped reads</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>2.3</b> Quantification</a></li>
<li class="chapter" data-level="2.4" data-path="exploratory-analysis.html"><a href="exploratory-analysis.html"><i class="fa fa-check"></i><b>2.4</b> Exploratory analysis</a></li>
<li class="chapter" data-level="2.5" data-path="understanding-the-dataset.html"><a href="understanding-the-dataset.html"><i class="fa fa-check"></i><b>2.5</b> Understanding the dataset</a></li>
<li class="chapter" data-level="2.6" data-path="prefiltering.html"><a href="prefiltering.html"><i class="fa fa-check"></i><b>2.6</b> Prefiltering</a></li>
<li class="chapter" data-level="2.7" data-path="normalisation.html"><a href="normalisation.html"><i class="fa fa-check"></i><b>2.7</b> Normalisation</a><ul>
<li class="chapter" data-level="2.7.1" data-path="normalisation.html"><a href="normalisation.html#defining-the-model-matrix"><i class="fa fa-check"></i><b>2.7.1</b> Defining the model matrix</a></li>
<li class="chapter" data-level="2.7.2" data-path="normalisation.html"><a href="normalisation.html#ma-plots"><i class="fa fa-check"></i><b>2.7.2</b> MA-plots</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html"><i class="fa fa-check"></i><b>2.8</b> Using visualisation to verify (sanity checks)</a><ul>
<li class="chapter" data-level="2.8.1" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>2.8.1</b> Principal Component Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>2.9</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.9.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#fitting-the-model"><i class="fa fa-check"></i><b>2.9.1</b> Fitting the model</a></li>
<li class="chapter" data-level="2.9.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#comparisons"><i class="fa fa-check"></i><b>2.9.2</b> Comparisons</a></li>
<li class="chapter" data-level="2.9.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#extract-top-de-features"><i class="fa fa-check"></i><b>2.9.3</b> Extract top DE features</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html"><i class="fa fa-check"></i><b>2.10</b> Verification using visualisation</a><ul>
<li class="chapter" data-level="2.10.1" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html#hierachical-clustering"><i class="fa fa-check"></i><b>2.10.1</b> Hierachical clustering</a></li>
</ul></li>
<li class="chapter" data-level="2.11" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>2.11</b> Gene Annotation</a></li>
<li class="chapter" data-level="2.12" data-path="gene-set-enrichment.html"><a href="gene-set-enrichment.html"><i class="fa fa-check"></i><b>2.12</b> Gene set enrichment</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differentially expressed gene analysis of RNA-seq data using R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-pre-processing" class="section level2">
<h2><span class="header-section-number">2.1</span> Data pre-processing</h2>
<div class="rmdexercise">
<p>
To start, create a new R project:
</p>
<ol style="list-style-type: decimal">
<li>
Click on <strong>File</strong> &gt; <strong>New Project</strong>
</li>
<li>
Click on <strong>New Directory</strong> &gt; <strong>Empty Project</strong>
</li>
<li>
Enter a name in <strong>Directory name</strong> (e.g.Â RNAseq_analysis). Click on <strong>Create Project</strong>. Wait for the project to be created and the page will refresh.
</li>
<li>
Check in the <strong>Files</strong> tab on the bottom-right corner. You should now see that you are in the <strong>Home &gt; RNAseq_analysis</strong> directory. This is the current working directory and in here, there should be a new file called <strong>RNAseq_analysis.Rproj</strong>.
</li>
</ol>
</div>
<p>The data considered for the RNAseq part of the workshop is BioProject PRJEB5297. It is available via ENA (<a href="http://www.ebi.ac.uk/ena/data/view/PRJEB5297" class="uri">http://www.ebi.ac.uk/ena/data/view/PRJEB5297</a>) or NCBI (<a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJEB5297" class="uri">https://www.ncbi.nlm.nih.gov/bioproject/PRJEB5297</a>). The study corresponds to 8 RNA sequencing libraries from Human brain and liver.</p>
<blockquote>
<p><em><strong>Allele-Specific Expression in Human Brain and Liver</strong></em> <em>Systematic survey of gene and isoform allele-specific expression in human brain and liver tissues, and description of optimised bioinformatic and statistical methods to accurately measure allele-specific expression.</em></p>
</blockquote>
<hr />
<p><strong>Our biological question for this study:</strong></p>
<p>What are the list of genes differentially expressed between the <span class="math inline">\(liver\)</span> and the <span class="math inline">\(brain\)</span> samples?</p>
<hr />
<p>Raw sequencing data are usually available in FASTQ format, which is a well defined text-based format for storing both biological sequences (usually nucleotide sequences) and their corresponding quality scores. The raw data from this study have been downloaded (8Gb / fastq file) into the shared directory <code>../data/RNAseq/raw_data</code>.</p>
<p>To see a list of files in this directory, enter the following commands:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">RNASeq_DATA_DIR &lt;-<span class="st"> &quot;../data/RNAseq/raw_data&quot;</span>
<span class="kw">dir</span>(RNASeq_DATA_DIR)</code></pre></div>
<pre><code>##  [1] &quot;ERR420386_1.fastq.gz&quot;         &quot;ERR420386_1.subset1.fastq.gz&quot;
##  [3] &quot;ERR420386_1.subset2.fastq.gz&quot; &quot;ERR420386_1.subset3.fastq.gz&quot;
##  [5] &quot;ERR420386_1.subset4.fastq.gz&quot; &quot;ERR420386_1.subset5.fastq.gz&quot;
##  [7] &quot;ERR420386_1.subset6.fastq.gz&quot; &quot;ERR420386_1.subset7.fastq.gz&quot;
##  [9] &quot;ERR420386_2.fastq.gz&quot;         &quot;ERR420386_chr21_R1.fastq.gz&quot; 
## [11] &quot;ERR420386_chr21_R2.fastq.gz&quot;  &quot;ERR420386_chr21.fastq.gz&quot;    
## [13] &quot;ERR420387_1.fastq.gz&quot;         &quot;ERR420387_2.fastq.gz&quot;        
## [15] &quot;ERR420388_1.fastq.gz&quot;         &quot;ERR420388_2.fastq.gz&quot;        
## [17] &quot;ERR420389_1.fastq.gz&quot;         &quot;ERR420389_2.fastq.gz&quot;        
## [19] &quot;ERR420390_1.fastq.gz&quot;         &quot;ERR420390_2.fastq.gz&quot;        
## [21] &quot;ERR420391_1.fastq.gz&quot;         &quot;ERR420391_2.fastq.gz&quot;        
## [23] &quot;ERR420392_1.fastq.gz&quot;         &quot;ERR420392_2.fastq.gz&quot;        
## [25] &quot;ERR420393_1.fastq.gz&quot;         &quot;ERR420393_2.fastq.gz&quot;        
## [27] &quot;experiment_design.txt&quot;</code></pre>
<p>The first step in a RNAseq analysis is to run a quick quality check on your data, this will give you an idea of the quality of your raw data in terms of number of reads per library, read length, average quality score along the reads, GC content, sequence duplication level, adaptors that might have not been removed correctly from the data etc.</p>
<p>The <code>fastQC</code> tool is quick and easy to run and can be downloaded from here: <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>.</p>
<p>To ensure highest quality of the sequences for subsequent mapping and differential expression analysis steps, the reads can also be trimmed using the <code>Trimmomatic</code> tool (Lohse et al. 2012).</p>
<div class="rmdnote">
<p>
For the scope of this course we will focus on the R-based steps and will assume that the data are fit for purpose.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">RNASeq_REF_DATA_DIR &lt;-<span class="st"> &quot;../data/RNAseq/ref_data&quot;</span>

REF_GENOME &lt;-<span class="st"> </span><span class="kw">file.path</span>(RNASeq_REF_DATA_DIR, <span class="st">&quot;hg19.fa&quot;</span>)
RSUBREAD_INDEX_PATH &lt;-<span class="st"> </span><span class="kw">file.path</span>(RNASeq_REF_DATA_DIR, <span class="st">&quot;RsubreadIndex&quot;</span>)

RSUBREAD_INDEX_BASE &lt;-<span class="st"> &quot;hg19&quot;</span>

RESULTS_DIR &lt;-<span class="st"> &#39;results/RNAseq&#39;</span>
MAPPING_DIR &lt;-<span class="st"> </span><span class="kw">file.path</span>(RESULTS_DIR,<span class="st">&#39;mapping&#39;</span>)
<span class="kw">dir.create</span>(MAPPING_DIR, <span class="dt">recursive=</span>T)</code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="rnaseq-data-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mapping.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-dataprep.Rmd",
"text": "Edit"
},
"download": ["RNAseq-analysis.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
