<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Differentially expressed gene analysis of RNA-seq data using R</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Differentially expressed gene analysis of RNA-seq data using R">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="using-visualisation-to-verify-sanity-checks.html">
<link rel="next" href="verification-using-visualisation.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">RNAseq analysis with R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> NGS data exploration</a><ul>
<li class="chapter" data-level="1.1" data-path="required-packages-for-these-exercises.html"><a href="required-packages-for-these-exercises.html"><i class="fa fa-check"></i><b>1.1</b> Required packages for these exercises</a></li>
<li class="chapter" data-level="1.2" data-path="reading-from-a-fasta-file.html"><a href="reading-from-a-fasta-file.html"><i class="fa fa-check"></i><b>1.2</b> Reading from a fasta file</a></li>
<li class="chapter" data-level="1.3" data-path="parsing-fastq-files.html"><a href="parsing-fastq-files.html"><i class="fa fa-check"></i><b>1.3</b> Parsing FastQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rnaseq-data-analysis.html"><a href="rnaseq-data-analysis.html"><i class="fa fa-check"></i><b>2</b> RNAseq data analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="mapping.html"><a href="mapping.html"><i class="fa fa-check"></i><b>2.2</b> Mapping</a><ul>
<li class="chapter" data-level="2.2.1" data-path="mapping.html"><a href="mapping.html#examining-the-mapped-reads"><i class="fa fa-check"></i><b>2.2.1</b> Examining the mapped reads</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>2.3</b> Quantification</a></li>
<li class="chapter" data-level="2.4" data-path="exploratory-analysis.html"><a href="exploratory-analysis.html"><i class="fa fa-check"></i><b>2.4</b> Exploratory analysis</a></li>
<li class="chapter" data-level="2.5" data-path="understanding-the-dataset.html"><a href="understanding-the-dataset.html"><i class="fa fa-check"></i><b>2.5</b> Understanding the dataset</a></li>
<li class="chapter" data-level="2.6" data-path="prefiltering.html"><a href="prefiltering.html"><i class="fa fa-check"></i><b>2.6</b> Prefiltering</a></li>
<li class="chapter" data-level="2.7" data-path="normalisation.html"><a href="normalisation.html"><i class="fa fa-check"></i><b>2.7</b> Normalisation</a><ul>
<li class="chapter" data-level="2.7.1" data-path="normalisation.html"><a href="normalisation.html#defining-the-model-matrix"><i class="fa fa-check"></i><b>2.7.1</b> Defining the model matrix</a></li>
<li class="chapter" data-level="2.7.2" data-path="normalisation.html"><a href="normalisation.html#ma-plots"><i class="fa fa-check"></i><b>2.7.2</b> MA-plots</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html"><i class="fa fa-check"></i><b>2.8</b> Using visualisation to verify (sanity checks)</a><ul>
<li class="chapter" data-level="2.8.1" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>2.8.1</b> Principal Component Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>2.9</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.9.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#fitting-the-model"><i class="fa fa-check"></i><b>2.9.1</b> Fitting the model</a></li>
<li class="chapter" data-level="2.9.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#comparisons"><i class="fa fa-check"></i><b>2.9.2</b> Comparisons</a></li>
<li class="chapter" data-level="2.9.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#extract-top-de-features"><i class="fa fa-check"></i><b>2.9.3</b> Extract top DE features</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html"><i class="fa fa-check"></i><b>2.10</b> Verification using visualisation</a><ul>
<li class="chapter" data-level="2.10.1" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html#hierachical-clustering"><i class="fa fa-check"></i><b>2.10.1</b> Hierachical clustering</a></li>
</ul></li>
<li class="chapter" data-level="2.11" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>2.11</b> Gene Annotation</a></li>
<li class="chapter" data-level="2.12" data-path="gene-set-enrichment.html"><a href="gene-set-enrichment.html"><i class="fa fa-check"></i><b>2.12</b> Gene set enrichment</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differentially expressed gene analysis of RNA-seq data using R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-expression-analysis" class="section level2">
<h2><span class="header-section-number">2.9</span> Differential expression analysis</h2>
<div id="fitting-the-model" class="section level3">
<h3><span class="header-section-number">2.9.1</span> Fitting the model</h3>
<p>To fit the model, use the <code>lmFit()</code> function, which takes in the normalised data object and the design matrix:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(y,design)</code></pre></div>
</div>
<div id="comparisons" class="section level3">
<h3><span class="header-section-number">2.9.2</span> Comparisons</h3>
<p>To have a look at the pairwise comparison(s) of interest, we may need to create a constrast matrix. This is usually always the case when there are multiple factors being compared. To do this, we use the <code>makeConstrasts()</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cont.matrix &lt;-<span class="st"> </span><span class="kw">makeContrasts</span>(liver-brain, <span class="dt">levels=</span>design)
cont.matrix</code></pre></div>
<pre><code>##        Contrasts
## Levels  liver - brain
##   brain            -1
##   liver             1</code></pre>
<p>Refit the model using the comparisons defined:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit2 &lt;-<span class="st"> </span><span class="kw">contrasts.fit</span>(fit, cont.matrix)
fit2 &lt;-<span class="st"> </span><span class="kw">eBayes</span>(fit2)
<span class="kw">options</span>(<span class="dt">digits =</span> <span class="dv">3</span>)

<span class="kw">dim</span>(fit2)</code></pre></div>
<pre><code>## [1] 15050     1</code></pre>
</div>
<div id="extract-top-de-features" class="section level3">
<h3><span class="header-section-number">2.9.3</span> Extract top DE features</h3>
<p>The <code>topTable</code> function summarises the output from limma in a table format. Significant DE genes for a particular comparison can be identified by selecting genes with a p-value smaller than a chosen cut-off value and/or a fold change greater than a chosen value in this table. By default the table will be sorted by increasing adjusted p-value, showing the most significant DE genes at the top.</p>
<p>Set the threshold values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">THRES_PVAL &lt;-<span class="st"> </span><span class="fl">0.01</span>
THRES_FC &lt;-<span class="st"> </span><span class="dv">3</span>

<span class="kw">colnames</span>(fit$coefficients)</code></pre></div>
<pre><code>## [1] &quot;brain&quot; &quot;liver&quot;</code></pre>
<p>Get the output table for the 10 most significant DE genes for this comparison: <span class="math inline">\(liver vs brain\)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">comparison=<span class="st">&#39;liver - brain&#39;</span>
<span class="kw">topTable</span>(fit2, <span class="dt">coef=</span>comparison)</code></pre></div>
<pre><code>##          logFC   AveExpr         t      P.Value    adj.P.Val        B
## 3240 10.818058  7.853145 100.71027 5.603924e-14 7.444180e-10 19.18750
## 5265 10.152815  7.487355  93.96364 9.892598e-14 7.444180e-10 19.01571
## 213  11.221081 10.966981  71.22374 9.577714e-13 2.059208e-09 18.47777
## 2335  6.797037  8.578822  67.73067 1.445862e-12 2.439635e-09 18.46460
## 338  11.093595  8.521278  71.76248 9.004441e-13 2.059208e-09 18.29472
## 2243 11.101339  7.402822  79.82534 3.763856e-13 1.461517e-09 18.09575
## 2244 11.177598  7.179453  75.38332 6.016490e-13 1.810963e-09 17.71017
## 229  10.583832  6.610364  79.51871 3.884431e-13 1.461517e-09 17.65905
## 125  10.915521  7.395653  67.65636 1.458918e-12 2.439635e-09 17.57556
## 716   7.466281  6.769933  59.39239 4.238953e-12 4.253083e-09 17.54615</code></pre>
<p>Get the full table (<code>n</code> is the number of genes in the fit):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">limma.result &lt;-<span class="st"> </span><span class="kw">topTable</span>(fit2, <span class="dt">coef=</span>comparison, <span class="dt">n=</span><span class="kw">nrow</span>(fit2))

## Get significant DEGs only (adjusted p-value &lt; THRES_PVAL)
limma.sigP.DEG &lt;-<span class="st"> </span><span class="kw">topTable</span>(fit2, <span class="dt">coef=</span>comparison, <span class="dt">n=</span><span class="kw">nrow</span>(fit2), <span class="dt">p.val=</span>THRES_PVAL)
<span class="kw">dim</span>(limma.sigP.DEG)</code></pre></div>
<pre><code>## [1] 8190    6</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(limma.sigP.DEG,<span class="dv">10</span>)</code></pre></div>
<pre><code>##          logFC   AveExpr         t      P.Value    adj.P.Val        B
## 3240 10.818058  7.853145 100.71027 5.603924e-14 7.444180e-10 19.18750
## 5265 10.152815  7.487355  93.96364 9.892598e-14 7.444180e-10 19.01571
## 213  11.221081 10.966981  71.22374 9.577714e-13 2.059208e-09 18.47777
## 2335  6.797037  8.578822  67.73067 1.445862e-12 2.439635e-09 18.46460
## 338  11.093595  8.521278  71.76248 9.004441e-13 2.059208e-09 18.29472
## 2243 11.101339  7.402822  79.82534 3.763856e-13 1.461517e-09 18.09575
## 2244 11.177598  7.179453  75.38332 6.016490e-13 1.810963e-09 17.71017
## 229  10.583832  6.610364  79.51871 3.884431e-13 1.461517e-09 17.65905
## 125  10.915521  7.395653  67.65636 1.458918e-12 2.439635e-09 17.57556
## 716   7.466281  6.769933  59.39239 4.238953e-12 4.253083e-09 17.54615</code></pre>
<p>Get significant DEG with low adjusted p-values and high fold change</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">limma.sigFC.DEG &lt;-<span class="st"> </span><span class="kw">subset</span>(limma.sigP.DEG, logFC &gt;<span class="st"> </span>THRES_FC)
<span class="kw">dim</span>(limma.sigFC.DEG)</code></pre></div>
<pre><code>## [1] 1291    6</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(limma.sigFC.DEG,<span class="dv">10</span>)</code></pre></div>
<pre><code>##          logFC   AveExpr         t      P.Value    adj.P.Val        B
## 3240 10.818058  7.853145 100.71027 5.603924e-14 7.444180e-10 19.18750
## 5265 10.152815  7.487355  93.96364 9.892598e-14 7.444180e-10 19.01571
## 213  11.221081 10.966981  71.22374 9.577714e-13 2.059208e-09 18.47777
## 2335  6.797037  8.578822  67.73067 1.445862e-12 2.439635e-09 18.46460
## 338  11.093595  8.521278  71.76248 9.004441e-13 2.059208e-09 18.29472
## 2243 11.101339  7.402822  79.82534 3.763856e-13 1.461517e-09 18.09575
## 2244 11.177598  7.179453  75.38332 6.016490e-13 1.810963e-09 17.71017
## 229  10.583832  6.610364  79.51871 3.884431e-13 1.461517e-09 17.65905
## 125  10.915521  7.395653  67.65636 1.458918e-12 2.439635e-09 17.57556
## 716   7.466281  6.769933  59.39239 4.238953e-12 4.253083e-09 17.54615</code></pre>
<p>Write the limma output table for significant genes to a tab-delimited file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filename &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;DEG_&quot;</span>,comparison,
                   <span class="st">&quot;_pval_&quot;</span>,THRES_PVAL,
                   <span class="st">&quot;_logFC_&quot;</span>, THRES_FC, <span class="st">&quot;.txt&quot;</span>);
<span class="kw">write.table</span>(limma.sigFC.DEG, <span class="dt">file=</span><span class="kw">file.path</span>(RESULTS_DIR,filename),
            <span class="dt">row.names=</span>T, <span class="dt">quote=</span>F, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)
filename</code></pre></div>
<pre><code>## [1] &quot;DEG_liver - brain_pval_0.01_logFC_3.txt&quot;</code></pre>
<div class="rmdexercise">
<p>
Get the number of DE genes between technical group 1 and technical group 2 (all Brain samples) with adj pvalue&lt;0.01
</p>
<ul>
<li>
Create a new design matrix for limma with the technical replicate groups
</li>
<li>
Re-normalise the read counts with ‘voom’ function with new design matrix
</li>
<li>
Fit a linear model on these normalised data
</li>
<li>
Make the contrast matrix corresponding to the new set of parameters
</li>
<li>
Fit the contrast matrix to the linear model
</li>
<li>
Compute moderated t-statistics of differential expression
</li>
<li>
Get the output table for the 10 most significant DE genes for this comparison
</li>
<li>
Get the number of genes significantly DE (adjusted p-value &lt; 0.01)
</li>
</ul>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="using-visualisation-to-verify-sanity-checks.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="verification-using-visualisation.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": {}
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["RNAseq-analysis.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
