<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Differentially expressed gene analysis of RNA-seq data using R</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Differentially expressed gene analysis of RNA-seq data using R">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="gene-annotation.html">


<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">RNAseq analysis with R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> NGS data exploration</a><ul>
<li class="chapter" data-level="1.1" data-path="required-packages-for-these-exercises.html"><a href="required-packages-for-these-exercises.html"><i class="fa fa-check"></i><b>1.1</b> Required packages for these exercises</a></li>
<li class="chapter" data-level="1.2" data-path="reading-from-a-fasta-file.html"><a href="reading-from-a-fasta-file.html"><i class="fa fa-check"></i><b>1.2</b> Reading from a fasta file</a></li>
<li class="chapter" data-level="1.3" data-path="parsing-fastq-files.html"><a href="parsing-fastq-files.html"><i class="fa fa-check"></i><b>1.3</b> Parsing FastQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rnaseq-data-analysis.html"><a href="rnaseq-data-analysis.html"><i class="fa fa-check"></i><b>2</b> RNAseq data analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="mapping.html"><a href="mapping.html"><i class="fa fa-check"></i><b>2.2</b> Mapping</a><ul>
<li class="chapter" data-level="2.2.1" data-path="mapping.html"><a href="mapping.html#examining-the-mapped-reads"><i class="fa fa-check"></i><b>2.2.1</b> Examining the mapped reads</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>2.3</b> Quantification</a></li>
<li class="chapter" data-level="2.4" data-path="exploratory-analysis.html"><a href="exploratory-analysis.html"><i class="fa fa-check"></i><b>2.4</b> Exploratory analysis</a></li>
<li class="chapter" data-level="2.5" data-path="understanding-the-dataset.html"><a href="understanding-the-dataset.html"><i class="fa fa-check"></i><b>2.5</b> Understanding the dataset</a></li>
<li class="chapter" data-level="2.6" data-path="prefiltering.html"><a href="prefiltering.html"><i class="fa fa-check"></i><b>2.6</b> Prefiltering</a></li>
<li class="chapter" data-level="2.7" data-path="normalisation.html"><a href="normalisation.html"><i class="fa fa-check"></i><b>2.7</b> Normalisation</a><ul>
<li class="chapter" data-level="2.7.1" data-path="normalisation.html"><a href="normalisation.html#defining-the-model-matrix"><i class="fa fa-check"></i><b>2.7.1</b> Defining the model matrix</a></li>
<li class="chapter" data-level="2.7.2" data-path="normalisation.html"><a href="normalisation.html#ma-plots"><i class="fa fa-check"></i><b>2.7.2</b> MA-plots</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html"><i class="fa fa-check"></i><b>2.8</b> Using visualisation to verify (sanity checks)</a><ul>
<li class="chapter" data-level="2.8.1" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>2.8.1</b> Principal Component Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>2.9</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.9.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#fitting-the-model"><i class="fa fa-check"></i><b>2.9.1</b> Fitting the model</a></li>
<li class="chapter" data-level="2.9.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#comparisons"><i class="fa fa-check"></i><b>2.9.2</b> Comparisons</a></li>
<li class="chapter" data-level="2.9.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#extract-top-de-features"><i class="fa fa-check"></i><b>2.9.3</b> Extract top DE features</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html"><i class="fa fa-check"></i><b>2.10</b> Verification using visualisation</a><ul>
<li class="chapter" data-level="2.10.1" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html#hierachical-clustering"><i class="fa fa-check"></i><b>2.10.1</b> Hierachical clustering</a></li>
</ul></li>
<li class="chapter" data-level="2.11" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>2.11</b> Gene Annotation</a></li>
<li class="chapter" data-level="2.12" data-path="gene-set-enrichment.html"><a href="gene-set-enrichment.html"><i class="fa fa-check"></i><b>2.12</b> Gene set enrichment</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differentially expressed gene analysis of RNA-seq data using R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gene-set-enrichment" class="section level2">
<h2><span class="header-section-number">2.12</span> Gene set enrichment</h2>
<p>Gene Ontology (GO) enrichment is a method for investigating sets of genes using the Gene Ontology system classification, in which genes are assigned to a particular set of terms for three major domains: cellular component, biological process and molecular function.</p>
<p>The <code>GOstats</code> package can test for both over and under representation of GO terms using the standard hypergeometric test. The output of the analysis is typically a ranked list of GO terms, each associated with a p-value.</p>
<p>The hypergeometric test will require both a list of selected genes (e.g. your DE genes) and a “universe” list (e.g. all genes represented that were tested for differential expression), all represented by their “EntrezGene” ID.</p>
<p>Get the list of universe entrez IDs:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GOstats)</code></pre></div>
<pre><code>## Loading required package: Category</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## 
## Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MASS&#39;:
## 
##     select</code></pre>
<pre><code>## Loading required package: Matrix</code></pre>
<pre><code>## 
## Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand</code></pre>
<pre><code>## The following object is masked from &#39;package:S4Vectors&#39;:
## 
##     expand</code></pre>
<pre><code>## Loading required package: graph</code></pre>
<pre><code>## 
## Attaching package: &#39;graph&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biostrings&#39;:
## 
##     complement</code></pre>
<pre><code>## </code></pre>
<pre><code>## 
## Attaching package: &#39;GOstats&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:AnnotationDbi&#39;:
## 
##     makeGOGraph</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">universe.entrezID &lt;-<span class="st"> </span><span class="kw">rownames</span>(filtered.raw.counts)
<span class="kw">length</span>(universe.entrezID)</code></pre></div>
<pre><code>## [1] 15050</code></pre>
<p>Before running the hypergeometric test with the <code>hyperGTest</code> function, you need to define the parameters for the test (gene lists, ontology, test direction) as well as the annotation database to be used. The ontology to be tested for can be any of the three GO domains: biological process (“BP”), cellular component (“CC”) or molecular function (“MF”).</p>
<p>In the example below, we will test for over-represented Biological Processes in our list of differentially expressed genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">annotationDB &lt;-<span class="st"> &quot;org.Hs.eg.db&quot;</span>
hgCutoff &lt;-<span class="st"> </span><span class="fl">0.05</span>

params &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;GOHyperGParams&quot;</span>,
              <span class="dt">geneIds=</span>DEG.entrezID,
              <span class="dt">universeGeneIds=</span>universe.entrezID,
              <span class="dt">annotation=</span>annotationDB,
              <span class="dt">ontology=</span><span class="st">&quot;BP&quot;</span>,
              <span class="dt">pvalueCutoff=</span>hgCutoff,
              <span class="dt">testDirection=</span><span class="st">&quot;over&quot;</span>)
<span class="co"># Run the test</span>
hg &lt;-<span class="st"> </span><span class="kw">hyperGTest</span>(params)
hg</code></pre></div>
<pre><code>## [1] 2010    7</code></pre>
<pre><code>##        GOBPID       Pvalue OddsRatio  ExpCount Count Size
## 1  GO:0006082 6.074927e-66  4.964910  73.47815   236  820
## 2  GO:0043436 1.146564e-58  4.766285  68.28092   216  762
## 3  GO:0019752 9.451120e-58  4.729580  67.92249   214  758
## 4  GO:0044281 1.899138e-54  3.336444 141.75907   326 1582
## 5  GO:0032787 9.530910e-54  5.891811  42.20513   159  471
## 6  GO:0006952 2.581186e-53  3.968852  88.26339   242  985
## 7  GO:0044710 5.216264e-51  2.692125 289.34263   509 3229
## 8  GO:0006954 1.371656e-48  5.643188  40.14416   148  448
## 9  GO:0006955 2.441550e-43  3.477192  91.04122   229 1016
## 10 GO:0008202 1.761494e-41  8.038462  20.16169    95  225
##                                     Term
## 1         organic acid metabolic process
## 2              oxoacid metabolic process
## 3      carboxylic acid metabolic process
## 4       small molecule metabolic process
## 5  monocarboxylic acid metabolic process
## 6                       defense response
## 7      single-organism metabolic process
## 8                  inflammatory response
## 9                        immune response
## 10             steroid metabolic process</code></pre>
<p>We need to adjust for multiple testing using the <code>p.adjust()</code> function. You can specify the type of adjustment method to use, we are using <code>bonferroni</code> in this example. Assign the adjusted pvalues back to the <code>hg.df</code> data object. Reorder the columns so that the unadjusted and adjusted p-values are next to each other.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hg.df &lt;-<span class="st"> </span><span class="kw">summary</span>(hg)
hg.df$Adj.Pvalue &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(hg.df$Pvalue, <span class="st">&#39;bonferroni&#39;</span>)
hg.df &lt;-<span class="st"> </span>hg.df[,<span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">2</span>,<span class="dv">8</span>,<span class="dv">3</span>:<span class="dv">7</span>)]
<span class="kw">head</span>(hg.df,<span class="dv">10</span>)</code></pre></div>
<pre><code>##        GOBPID       Pvalue   Adj.Pvalue OddsRatio  ExpCount Count Size
## 1  GO:0006082 6.074927e-66 1.221060e-62  4.964910  73.47815   236  820
## 2  GO:0043436 1.146564e-58 2.304594e-55  4.766285  68.28092   216  762
## 3  GO:0019752 9.451120e-58 1.899675e-54  4.729580  67.92249   214  758
## 4  GO:0044281 1.899138e-54 3.817267e-51  3.336444 141.75907   326 1582
## 5  GO:0032787 9.530910e-54 1.915713e-50  5.891811  42.20513   159  471
## 6  GO:0006952 2.581186e-53 5.188184e-50  3.968852  88.26339   242  985
## 7  GO:0044710 5.216264e-51 1.048469e-47  2.692125 289.34263   509 3229
## 8  GO:0006954 1.371656e-48 2.757028e-45  5.643188  40.14416   148  448
## 9  GO:0006955 2.441550e-43 4.907515e-40  3.477192  91.04122   229 1016
## 10 GO:0008202 1.761494e-41 3.540603e-38  8.038462  20.16169    95  225
##                                     Term
## 1         organic acid metabolic process
## 2              oxoacid metabolic process
## 3      carboxylic acid metabolic process
## 4       small molecule metabolic process
## 5  monocarboxylic acid metabolic process
## 6                       defense response
## 7      single-organism metabolic process
## 8                  inflammatory response
## 9                        immune response
## 10             steroid metabolic process</code></pre>
<p>Compare before and after multiple adjustment:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(<span class="kw">density</span>(hg.df$Pvalue),<span class="st">&#39;Unadjust p-values&#39;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>hgCutoff,<span class="dt">col=</span><span class="st">&#39;red&#39;</span>,<span class="dt">lty=</span><span class="dv">2</span>)

<span class="kw">plot</span>(<span class="kw">density</span>(hg.df$Adj.Pvalue),<span class="st">&#39;Adjusted p-values&#39;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>hgCutoff,<span class="dt">col=</span><span class="st">&#39;red&#39;</span>,<span class="dt">lty=</span><span class="dv">2</span>)</code></pre></div>
<p><img src="RNAseq-analysis_files/figure-html/GO-plot-1.png" width="672" /></p>
<p>Keep only the significant GO terms after adjusting for multiple testing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sigGO.table &lt;-<span class="st"> </span><span class="kw">subset</span>(hg.df, Adj.Pvalue &lt;<span class="st"> </span>hgCutoff)
<span class="kw">dim</span>(sigGO.table)</code></pre></div>
<pre><code>## [1] 517   8</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(sigGO.table,<span class="dv">10</span>)</code></pre></div>
<pre><code>##        GOBPID       Pvalue   Adj.Pvalue OddsRatio  ExpCount Count Size
## 1  GO:0006082 6.074927e-66 1.221060e-62  4.964910  73.47815   236  820
## 2  GO:0043436 1.146564e-58 2.304594e-55  4.766285  68.28092   216  762
## 3  GO:0019752 9.451120e-58 1.899675e-54  4.729580  67.92249   214  758
## 4  GO:0044281 1.899138e-54 3.817267e-51  3.336444 141.75907   326 1582
## 5  GO:0032787 9.530910e-54 1.915713e-50  5.891811  42.20513   159  471
## 6  GO:0006952 2.581186e-53 5.188184e-50  3.968852  88.26339   242  985
## 7  GO:0044710 5.216264e-51 1.048469e-47  2.692125 289.34263   509 3229
## 8  GO:0006954 1.371656e-48 2.757028e-45  5.643188  40.14416   148  448
## 9  GO:0006955 2.441550e-43 4.907515e-40  3.477192  91.04122   229 1016
## 10 GO:0008202 1.761494e-41 3.540603e-38  8.038462  20.16169    95  225
##                                     Term
## 1         organic acid metabolic process
## 2              oxoacid metabolic process
## 3      carboxylic acid metabolic process
## 4       small molecule metabolic process
## 5  monocarboxylic acid metabolic process
## 6                       defense response
## 7      single-organism metabolic process
## 8                  inflammatory response
## 9                        immune response
## 10             steroid metabolic process</code></pre>
<p>Other software can be used to investigate over-represented pathways, such as GeneGO <a href="https://portal.genego.com/" class="uri">https://portal.genego.com/</a> and Ingenuity <a href="http://www.ingenuity.com/products/" class="uri">http://www.ingenuity.com/products/</a> ipa. The advantage of these applications is that they maintain curated and up-to-date extensive databases. They also provide intuitive visualisation and network modelling tools.</p>
<hr />
<p>Save an image of your RNAseq analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">save.image</span>(<span class="dt">file=</span><span class="kw">file.path</span>(RESULTS_DIR,<span class="st">&quot;RNAseq.Rdata&quot;</span>))</code></pre></div>
<p></p>
<p><strong>SessionInfo</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.3.3 (2017-03-06)
## Platform: x86_64-redhat-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## locale:
##  [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
##  [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
##  [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] biomaRt_2.30.0             tidyr_0.6.1               
##  [3] edgeR_3.16.5               limma_3.30.13             
##  [5] mixOmics_6.1.2             ggplot2_2.2.1             
##  [7] lattice_0.20-35            MASS_7.3-45               
##  [9] Rsubread_1.24.2            ShortRead_1.32.1          
## [11] GenomicAlignments_1.10.1   SummarizedExperiment_1.4.0
## [13] Biobase_2.34.0             Rsamtools_1.26.2          
## [15] GenomicRanges_1.26.4       GenomeInfoDb_1.10.3       
## [17] BiocParallel_1.8.2         Biostrings_2.42.1         
## [19] XVector_0.14.1             IRanges_2.8.2             
## [21] S4Vectors_0.12.2           BiocGenerics_0.20.0       
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.10         locfit_1.5-9.1       corpcor_1.6.9       
##  [4] assertthat_0.2.0     rprojroot_1.2        digest_0.6.12       
##  [7] mime_0.5             R6_2.2.0             plyr_1.8.4          
## [10] backports_1.0.5      RSQLite_1.1-2        ellipse_0.3-8       
## [13] evaluate_0.10        zlibbioc_1.20.0      lazyeval_0.2.0      
## [16] rstudioapi_0.6       Matrix_1.2-8         rmarkdown_1.4       
## [19] labeling_0.3         stringr_1.2.0        htmlwidgets_0.8     
## [22] igraph_1.0.1         RCurl_1.95-4.8       munsell_0.4.3       
## [25] shiny_1.0.1          httpuv_1.3.3         htmltools_0.3.5     
## [28] tibble_1.3.0         bookdown_0.3         codetools_0.2-15    
## [31] XML_3.98-1.6         dplyr_0.5.0          bitops_1.0-6        
## [34] grid_3.3.3           jsonlite_1.4         xtable_1.8-2        
## [37] gtable_0.2.0         DBI_0.6-1            magrittr_1.5        
## [40] scales_0.4.1         stringi_1.1.5        hwriter_1.3.2       
## [43] reshape2_1.4.2       latticeExtra_0.6-28  RColorBrewer_1.1-2  
## [46] tools_3.3.3          yaml_2.1.14          AnnotationDbi_1.36.2
## [49] colorspace_1.3-2     memoise_1.0.0        knitr_1.15.1</code></pre>

</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="gene-annotation.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>


<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": {}
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["RNAseq-analysis.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
