<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Differentially expressed gene analysis of RNA-seq data using R</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Differentially expressed gene analysis of RNA-seq data using R">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="quantification.html">
<link rel="next" href="checking-for-batch-or-confounding-effects.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> NGS data exploration</a><ul>
<li class="chapter" data-level="1.1" data-path="required-packages-for-these-exercises.html"><a href="required-packages-for-these-exercises.html"><i class="fa fa-check"></i><b>1.1</b> Required packages for these exercises</a></li>
<li class="chapter" data-level="1.2" data-path="reading-from-a-fasta-file.html"><a href="reading-from-a-fasta-file.html"><i class="fa fa-check"></i><b>1.2</b> Reading from a fasta file</a></li>
<li class="chapter" data-level="1.3" data-path="parsing-fastq-files.html"><a href="parsing-fastq-files.html"><i class="fa fa-check"></i><b>1.3</b> Parsing FastQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rnaseq-data-analysis.html"><a href="rnaseq-data-analysis.html"><i class="fa fa-check"></i><b>2</b> RNAseq data analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="mapping.html"><a href="mapping.html"><i class="fa fa-check"></i><b>2.2</b> Mapping</a><ul>
<li class="chapter" data-level="2.2.1" data-path="mapping.html"><a href="mapping.html#examining-the-mapped-reads"><i class="fa fa-check"></i><b>2.2.1</b> Examining the mapped reads</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>2.3</b> Quantification</a></li>
<li class="chapter" data-level="2.4" data-path="quality-checking.html"><a href="quality-checking.html"><i class="fa fa-check"></i><b>2.4</b> Quality checking</a></li>
<li class="chapter" data-level="2.5" data-path="checking-for-batch-or-confounding-effects.html"><a href="checking-for-batch-or-confounding-effects.html"><i class="fa fa-check"></i><b>2.5</b> Checking for batch or confounding effects</a></li>
<li class="chapter" data-level="2.6" data-path="prefiltering.html"><a href="prefiltering.html"><i class="fa fa-check"></i><b>2.6</b> Prefiltering</a></li>
<li class="chapter" data-level="2.7" data-path="normalisation.html"><a href="normalisation.html"><i class="fa fa-check"></i><b>2.7</b> Normalisation</a></li>
<li class="chapter" data-level="2.8" data-path="batch-and-sanity-checks.html"><a href="batch-and-sanity-checks.html"><i class="fa fa-check"></i><b>2.8</b> Batch and sanity checks</a><ul>
<li class="chapter" data-level="2.8.1" data-path="batch-and-sanity-checks.html"><a href="batch-and-sanity-checks.html#hierachical-clustering"><i class="fa fa-check"></i><b>2.8.1</b> Hierachical clustering</a></li>
<li class="chapter" data-level="2.8.2" data-path="batch-and-sanity-checks.html"><a href="batch-and-sanity-checks.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>2.8.2</b> Principal Component Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>2.9</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.9.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#model-matrices"><i class="fa fa-check"></i><b>2.9.1</b> Model matrices</a></li>
<li class="chapter" data-level="2.9.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#ma-plots"><i class="fa fa-check"></i><b>2.9.2</b> MA-plots</a></li>
<li class="chapter" data-level="2.9.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#fitting-the-model"><i class="fa fa-check"></i><b>2.9.3</b> Fitting the model</a></li>
<li class="chapter" data-level="2.9.4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#comparisons"><i class="fa fa-check"></i><b>2.9.4</b> Comparisons</a></li>
<li class="chapter" data-level="2.9.5" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#extract-top-de-features"><i class="fa fa-check"></i><b>2.9.5</b> Extract top DE features</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="verification.html"><a href="verification.html"><i class="fa fa-check"></i><b>2.10</b> Verification</a></li>
<li class="chapter" data-level="2.11" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>2.11</b> Gene Annotation</a></li>
<li class="chapter" data-level="2.12" data-path="gene-set-enrichment.html"><a href="gene-set-enrichment.html"><i class="fa fa-check"></i><b>2.12</b> Gene set enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sessioninfo.html"><a href="sessioninfo.html"><i class="fa fa-check"></i><b>3</b> SessionInfo</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differentially expressed gene analysis of RNA-seq data using R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-checking" class="section level2">
<h2><span class="header-section-number">2.4</span> Quality checking</h2>
<p>Rsubread provides the number of reads mapped to each gene which can then be used for ploting quality control figures and for differential expression analysis.</p>
<p>QC figures of the mapped read counts can be plotted and investigated for potential outlier libraries and to confirm grouping of samples.</p>
<p>Before plotting QC figures it is useful to get the experiment design. This will allow labeling of the data with the sample groups they belong to, or any other parameter of interest.</p>
<p>The experiment design file corresponding to this study has been downloaded from the ArrayExpress webpage and formatted as a tab separated file for this analysis purposes. You can find it in the shared directory <code>../data/Data_Analysis_with_R/RNAseq/raw_data</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">EXPMT_DESIGN_FILE &lt;-<span class="st"> </span><span class="kw">file.path</span>(RNASeq_DATA_DIR, <span class="st">&#39;experiment_design.txt&#39;</span>)

expr.design &lt;-<span class="st"> </span><span class="kw">read.table</span>(EXPMT_DESIGN_FILE, <span class="dt">header=</span>T, <span class="dt">sep=</span><span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>)
<span class="kw">rownames</span>(expr.design) &lt;-<span class="st"> </span>expr.design$SampleID

<span class="co">#order the design in the same ordering as the counts object</span>
expr.design &lt;-<span class="st"> </span>expr.design[<span class="kw">colnames</span>(counts$counts),]

expr.design</code></pre></div>
<pre><code>##            SampleID    Source.Name     organism  sex age tissue
## ERR420386 ERR420386 brain_sample_1 Homo sapiens male  26  brain
## ERR420387 ERR420387 brain_sample_1 Homo sapiens male  26  brain
## ERR420388 ERR420388 liver_sample_1 Homo sapiens male  30  liver
## ERR420389 ERR420389 liver_sample_1 Homo sapiens male  30  liver
## ERR420390 ERR420390 liver_sample_1 Homo sapiens male  30  liver
## ERR420391 ERR420391 brain_sample_1 Homo sapiens male  26  brain
## ERR420392 ERR420392 brain_sample_1 Homo sapiens male  26  brain
## ERR420393 ERR420393 liver_sample_1 Homo sapiens male  30  liver
##           Extract.Name Material.Type Assay.Name technical.replicate.group
## ERR420386       GCCAAT           RNA     Assay4                   group_2
## ERR420387       ACAGTG           RNA     Assay2                   group_1
## ERR420388       GTGAAA           RNA     Assay7                   group_4
## ERR420389       GTGAAA           RNA     Assay8                   group_4
## ERR420390       CTTGTA           RNA     Assay6                   group_3
## ERR420391       ACAGTG           RNA     Assay1                   group_1
## ERR420392       GCCAAT           RNA     Assay3                   group_2
## ERR420393       CTTGTA           RNA     Assay5                   group_3</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samples &lt;-<span class="st"> </span><span class="kw">as.character</span>(expr.design$SampleID)
group &lt;-<span class="st"> </span><span class="kw">factor</span>(expr.design$tissue)
group</code></pre></div>
<pre><code>## [1] brain brain liver liver liver brain brain liver
## Levels: brain liver</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">age &lt;-<span class="st"> </span><span class="kw">factor</span>(expr.design$age)
age</code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="quantification.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="checking-for-batch-or-confounding-effects.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-QC_stats.Rmd",
"text": "Edit"
},
"download": ["RNAseq-analysis.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
