<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Differentially expressed gene analysis of RNA-seq data using R</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Differentially expressed gene analysis of RNA-seq data using R">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Differentially expressed gene analysis of RNA-seq data using R" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="mapping.html">
<link rel="next" href="exploratory-analysis.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> NGS data exploration</a><ul>
<li class="chapter" data-level="1.1" data-path="required-packages-for-these-exercises.html"><a href="required-packages-for-these-exercises.html"><i class="fa fa-check"></i><b>1.1</b> Required packages for these exercises</a></li>
<li class="chapter" data-level="1.2" data-path="reading-from-a-fasta-file.html"><a href="reading-from-a-fasta-file.html"><i class="fa fa-check"></i><b>1.2</b> Reading from a fasta file</a></li>
<li class="chapter" data-level="1.3" data-path="parsing-fastq-files.html"><a href="parsing-fastq-files.html"><i class="fa fa-check"></i><b>1.3</b> Parsing FastQ files</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rnaseq-data-analysis.html"><a href="rnaseq-data-analysis.html"><i class="fa fa-check"></i><b>2</b> RNAseq data analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="data-pre-processing.html"><a href="data-pre-processing.html"><i class="fa fa-check"></i><b>2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2" data-path="mapping.html"><a href="mapping.html"><i class="fa fa-check"></i><b>2.2</b> Mapping</a><ul>
<li class="chapter" data-level="2.2.1" data-path="mapping.html"><a href="mapping.html#examining-the-mapped-reads"><i class="fa fa-check"></i><b>2.2.1</b> Examining the mapped reads</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="quantification.html"><a href="quantification.html"><i class="fa fa-check"></i><b>2.3</b> Quantification</a></li>
<li class="chapter" data-level="2.4" data-path="exploratory-analysis.html"><a href="exploratory-analysis.html"><i class="fa fa-check"></i><b>2.4</b> Exploratory analysis</a></li>
<li class="chapter" data-level="2.5" data-path="understanding-the-dataset.html"><a href="understanding-the-dataset.html"><i class="fa fa-check"></i><b>2.5</b> Understanding the dataset</a></li>
<li class="chapter" data-level="2.6" data-path="prefiltering.html"><a href="prefiltering.html"><i class="fa fa-check"></i><b>2.6</b> Prefiltering</a></li>
<li class="chapter" data-level="2.7" data-path="normalisation.html"><a href="normalisation.html"><i class="fa fa-check"></i><b>2.7</b> Normalisation</a><ul>
<li class="chapter" data-level="2.7.1" data-path="normalisation.html"><a href="normalisation.html#defining-the-model-matrix"><i class="fa fa-check"></i><b>2.7.1</b> Defining the model matrix</a></li>
<li class="chapter" data-level="2.7.2" data-path="normalisation.html"><a href="normalisation.html#ma-plots"><i class="fa fa-check"></i><b>2.7.2</b> MA-plots</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html"><i class="fa fa-check"></i><b>2.8</b> Using visualisation to verify (sanity checks)</a><ul>
<li class="chapter" data-level="2.8.1" data-path="using-visualisation-to-verify-sanity-checks.html"><a href="using-visualisation-to-verify-sanity-checks.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>2.8.1</b> Principal Component Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>2.9</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.9.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#fitting-the-model"><i class="fa fa-check"></i><b>2.9.1</b> Fitting the model</a></li>
<li class="chapter" data-level="2.9.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#comparisons"><i class="fa fa-check"></i><b>2.9.2</b> Comparisons</a></li>
<li class="chapter" data-level="2.9.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#extract-top-de-features"><i class="fa fa-check"></i><b>2.9.3</b> Extract top DE features</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html"><i class="fa fa-check"></i><b>2.10</b> Verification using visualisation</a><ul>
<li class="chapter" data-level="2.10.1" data-path="verification-using-visualisation.html"><a href="verification-using-visualisation.html#hierachical-clustering"><i class="fa fa-check"></i><b>2.10.1</b> Hierachical clustering</a></li>
</ul></li>
<li class="chapter" data-level="2.11" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>2.11</b> Gene Annotation</a></li>
<li class="chapter" data-level="2.12" data-path="gene-set-enrichment.html"><a href="gene-set-enrichment.html"><i class="fa fa-check"></i><b>2.12</b> Gene set enrichment</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differentially expressed gene analysis of RNA-seq data using R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quantification" class="section level2">
<h2><span class="header-section-number">2.3</span> Quantification</h2>
<p>Rsubread provides a read summarization function <code>featureCounts</code>, which takes two inputs:</p>
<ol style="list-style-type: decimal">
<li>the aligned reads (BAM or SAM) and assigns them to</li>
<li>genomic features (GTF annotation file)</li>
</ol>
<p>This gives the number of reads mapped per feature, which can then be normalised and tested for differential expression.</p>
<p>Rsubread comes with in-built annotations for <span class="math inline">\(mm9\)</span>, <span class="math inline">\(mm10\)</span> and <span class="math inline">\(hg19\)</span> for users’ convenience, but you can also supply your own annotation file (GTF), see the tip below.</p>
<div class="rmdwarning">
<p>
For experiments with lots of samples or on big genomes, this step can also take a while. We will only be performing the feature counts on the subset <code>chr21</code> BAM file we created previously.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mini.counts &lt;-<span class="st"> </span><span class="kw">featureCounts</span>(output.bamFile, 
                        <span class="dt">annot.inbuilt=</span><span class="st">&quot;hg19&quot;</span>,
                        <span class="dt">isGTFAnnotationFile=</span><span class="ot">FALSE</span>, 
                        <span class="dt">isPairedEnd=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## NCBI RefSeq annotation for hg19 (build 37.2) is used.
## 
##         ==========     _____ _    _ ____  _____  ______          _____  
##         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ 
##           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
##             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |
##               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
##         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
##        Rsubread 1.24.2
## 
## //========================== featureCounts setting ===========================\\
## ||                                                                            ||
## ||             Input files : 1 BAM file                                       ||
## ||                           P results/RNAseq/mapping/ERR420386.bam           ||
## ||                                                                            ||
## ||      Dir for temp files : .                                                ||
## ||                 Threads : 1                                                ||
## ||                   Level : meta-feature level                               ||
## ||              Paired-end : yes                                              ||
## ||         Strand specific : no                                               ||
## ||      Multimapping reads : not counted                                      ||
## || Multi-overlapping reads : not counted                                      ||
## ||   Min overlapping bases : 1                                                ||
## ||                                                                            ||
## ||          Chimeric reads : counted                                          ||
## ||        Both ends mapped : not required                                     ||
## ||                                                                            ||
## \\===================== http://subread.sourceforge.net/ ======================//
## 
## //================================= Running ==================================\\
## ||                                                                            ||
## || Load annotation file /usr/lib64/R/library/Rsubread/annot/hg19_RefSeq_e ... ||
## ||    Features : 225074                                                       ||
## ||    Meta-features : 25702                                                   ||
## ||    Chromosomes/contigs : 52                                                ||
## ||                                                                            ||
## || Process BAM file results/RNAseq/mapping/ERR420386.bam...                   ||
## ||    Paired-end reads are included.                                          ||
## ||    Assign fragments (read pairs) to features...                            ||
## ||    Total fragments : 161297                                                ||
## ||    Successfully assigned fragments : 72440 (44.9%)                         ||
## ||    Running time : 0.01 minutes                                             ||
## ||                                                                            ||
## ||                         Read assignment finished.                          ||
## ||                                                                            ||
## \\===================== http://subread.sourceforge.net/ ======================//</code></pre>
<p>Examine the attributes in the returned <code>mini.counts</code> object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(mini.counts)</code></pre></div>
<pre><code>##            Length Class      Mode     
## counts     25702  -none-     numeric  
## annotation     6  data.frame list     
## targets        1  -none-     character
## stat           2  data.frame list</code></pre>
<p>Find the dimension of the <code>counts</code> table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(mini.counts$counts)</code></pre></div>
<pre><code>## [1] 25702     1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mini.counts$counts[<span class="dv">1</span>:<span class="dv">6</span>,]</code></pre></div>
<pre><code>##    653635 100422834    645520     79501    729737 100507658 
##         0         0         0         0         0         0</code></pre>
<p>Look at the annotations, which corresponds to the rows of the <code>counts</code> table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(mini.counts$annotation)</code></pre></div>
<pre><code>##      GeneID                                                    Chr
## 1    653635 chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1;chr1
## 2 100422834                                                   chr1
## 3    645520                                         chr1;chr1;chr1
## 4     79501                                                   chr1
## 5    729737                                    chr1;chr1;chr1;chr1
## 6 100507658                                         chr1;chr1;chr1
##                                                               Start
## 1 14362;14970;15796;16607;16858;17233;17606;17915;18268;24738;29321
## 2                                                             30366
## 3                                                 34611;35277;35721
## 4                                                             69091
## 5                                       136698;136953;139790;140075
## 6                                              319944;320881;321032
##                                                                 End
## 1 14829;15038;15947;16765;17055;17368;17742;18061;18366;24891;29370
## 2                                                             30503
## 3                                                 35174;35481;36081
## 4                                                             70008
## 5                                       136805;139696;139847;140566
## 6                                              320653;320938;321056
##                  Strand Length
## 1 -;-;-;-;-;-;-;-;-;-;-   1769
## 2                     +    138
## 3                 -;-;-   1130
## 4                     +    918
## 5               -;-;-;-   3402
## 6                 +;+;+    793</code></pre>
<p><code>featureCounts</code> also returns a very hand summary of the number of reads that were <em>assigned</em> or <em>unassiged</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mini.counts$stat</code></pre></div>
<pre><code>##                       Status results.RNAseq.mapping.ERR420386.bam
## 1                   Assigned                                72440
## 2       Unassigned_Ambiguity                                  227
## 3    Unassigned_MultiMapping                                    0
## 4      Unassigned_NoFeatures                                87851
## 5        Unassigned_Unmapped                                  779
## 6  Unassigned_MappingQuality                                    0
## 7  Unassigned_FragmentLength                                    0
## 8         Unassigned_Chimera                                    0
## 9       Unassigned_Secondary                                    0
## 10    Unassigned_Nonjunction                                    0
## 11      Unassigned_Duplicate                                    0</code></pre>
<div class="rmdexercise">
<ol style="list-style-type: decimal">
<li>
<p>
Run the above command to perform counting reads for the bam file and then take a look at the summary output.
</p>
</li>
<li>
<p>
Lookup the user guide for Rsubread and find the defintions for the status in <code>mini.counts$stat</code> table, specifically <code>Unassigned_Ambiguity</code>, <code>Unassigned_NoFeatures</code> and <code>Unassigned_Unmapped</code>.
</p>
</li>
</ol>
</div>
<div class="rmdtip">
<p>
If you want to get read counts using another annotation.GTF file, use the <code>annot.ext</code> parameter. For example: counts &lt;- featureCounts(output.bamFile, annot.ext=“annotation.GTF”, isGTFAnnotationFile=TRUE, isPairedEnd=TRUE)
</p>
</div>
<div class="rmdwarning">
<p>
For the purpose of this workshop the read summarisation step has already been performed for all libraries. You will need to load the corresponding <code>Rdata</code> file to get these read counts. You can then print out these counts in a text file for future use.
</p>
</div>
<p>To load the Rdata object file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;../data/RNAseq/quantification/rawCounts.Rdata&quot;</span>)
<span class="kw">summary</span>(counts)</code></pre></div>
<pre><code>##            Length Class      Mode     
## counts     205616 -none-     numeric  
## annotation      6 data.frame list     
## targets         8 -none-     character
## stat            9 data.frame list</code></pre>
<p>Since we will be using it later for DE analysis, let’s create a <code>raw.counts</code> object to hold only the count data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">raw.counts &lt;-<span class="st"> </span>counts$counts
<span class="kw">dim</span>(raw.counts)</code></pre></div>
<pre><code>## [1] 25702     8</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">raw.counts[<span class="dv">1</span>:<span class="dv">6</span>,<span class="dv">1</span>:<span class="dv">4</span>]</code></pre></div>
<pre><code>##           ERR420386 ERR420387 ERR420388 ERR420389
## 653635           47       151        93       112
## 100422834         1         0         0         0
## 645520            1         2         0         0
## 79501             0         0         0         0
## 729737           15        35        47        57
## 100507658        10         5        15         8</code></pre>
<div class="rmdexercise">
<p>
<strong>Challenge</strong>
</p>
<ol style="list-style-type: decimal">
<li>
<p>
What are the count statistics for the “Brain and Liver” study?
</p>
</li>
<li>
<p>
create a copy of the stats object and change the counts into proportion per sample.
</p>
</li>
<li>
<p>
The <code>stat</code> information from <code>featureCounts()</code> will be much easier to digest if we can plot the proportion (or number) of reads assigned and unassigned due to the different status type. Can you create the following plot?
</p>
</li>
</ol>
</div>
<p><img src="RNAseq-analysis_files/figure-html/full-counts-1.png" width="768" /></p>
<div class="rmdsolution">
<p>
<strong>Solution</strong>
</p>
<p>
All lines starting with <code>##</code> are comments.
</p>
</div>
<p>Export out the counts table for every sample into a tab-separated file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">write.table</span>(counts$counts, <span class="dt">file=</span><span class="kw">file.path</span>(RESULTS_DIR,<span class="st">&quot;raw_read_counts.txt&quot;</span>), 
            <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span>F, <span class="dt">append=</span>F)</code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="mapping.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="exploratory-analysis.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-mapping.Rmd",
"text": "Edit"
},
"download": ["RNAseq-analysis.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
